runtime/src/iree-amd-aie/aie_runtime/iree_aie_router.cc

```diff
@@ -211,114 +463,498 @@ std::optional<std::map<PathEndPoint, SwitchSettings>> Router::findPaths(
+            shimCh = channel == 0 ? 2 : 3;
+            addConnection(curr, StrmSwPortType::NORTH, shimCh, bundle, channel,
+                          shimMuxOp.first, shimMuxOp.second);
+          } else if (bundle == StrmSwPortType::NOC) {
+            // must be either N2/3/4/5 -> NOC0/1/2/3
+            shimCh = channel + 2;
+            addConnection(curr, StrmSwPortType::NORTH, shimCh, bundle, channel,
+                          shimMuxOp.first, shimMuxOp.second);
+          }
+        }
+        addConnection(curr, setting.src.bundle, setting.src.channel,
+                      StrmSwPortType::SOUTH, shimCh, std::get<0>(swOp),
+                      std::get<1>(swOp), std::get<2>(swOp));
+      } else {
+        // otherwise, regular switchbox connection
+        addConnection(curr, setting.src.bundle, setting.src.channel, bundle,
+                      channel, std::get<0>(swOp), std::get<1>(swOp),
+                      std::get<2>(swOp));
+      }
+    }
+  }
+
+  return connections;
+}
+
+bool existsPathToDest(const SwitchSettings &settings, TileLoc currTile,
+                      StrmSwPortType currDestBundle, int currDestChannel,
+                      TileLoc finalTile, StrmSwPortType finalDestBundle,
+                      int finalDestChannel) {
+  if ((currTile == finalTile) && (currDestBundle == finalDestBundle) &&
+      (currDestChannel == finalDestChannel)) {
+    return true;
+  }
+
+  StrmSwPortType neighbourSourceBundle;
+  TileLoc neighbourTile{-1, -1};
+  if (currDestBundle == StrmSwPortType::EAST) {
+    neighbourSourceBundle = StrmSwPortType::WEST;
+    neighbourTile = {currTile.col + 1, currTile.row};
+  } else if (currDestBundle == StrmSwPortType::WEST) {
+    neighbourSourceBundle = StrmSwPortType::EAST;
+    neighbourTile = {currTile.col - 1, currTile.row};
+  } else if (currDestBundle == StrmSwPortType::NORTH) {
+    neighbourSourceBundle = StrmSwPortType::SOUTH;
+    neighbourTile = {currTile.col, currTile.row + 1};
+  } else if (currDestBundle == StrmSwPortType::SOUTH) {
+    neighbourSourceBundle = StrmSwPortType::NORTH;
+    neighbourTile = {currTile.col, currTile.row - 1};
+  } else {
+    return false;
+  }
+  assert(neighbourTile.col != -1 && neighbourTile.row != -1);
+
+  int neighbourSourceChannel = currDestChannel;
+  for (const auto &[sbNode, setting] : settings) {
+    TileLoc tile = {sbNode.col, sbNode.row};
+    if ((tile == neighbourTile) &&
+        (setting.src.bundle == neighbourSourceBundle) &&
+        (setting.src.channel == neighbourSourceChannel)) {
+      for (const auto &[bundle, channel] : setting.dsts) {
+        if (existsPathToDest(settings, neighbourTile, bundle, channel,
+                             finalTile, finalDestBundle, finalDestChannel)) {
+          return true;
+        }
+      }
+    }
+  }
+
+  return false;
+}
+
+std::tuple<MasterSetsT, SlaveGroupsT, SlaveMasksT, SlaveAMSelsT>
+configurePacketFlows(int numMsels, int numArbiters,
```

Comment:
I wish I understood this completely but the reason I didn't add a docstring here even though I knew you would ask is because I don't know how packet routing works (not really). I can add a kind of "play-by-play" if you really want but it's a bad idea (like a blind man describing a soccer match...).

Reply:
Sure, I am wondering what 'configuring' actually means here? It won't be the actual configuring (using `aie-rt`), so I was wondering what exactly

Reply:
just emitting those structs/DenseMaps/whatever. I'll put that docstring on it.